### ISSUE TITLE
Problems with Ubuntu 19.04

### ISSUE THREAD

**Issue by heiderich** [openwebwork/webwork2#938]
I experience problems running the current master branch (db7eb3b37b01a781c4bcade342ddbf79a292cd3b) of webwork2 on Ubuntu 19.04 using docker.

The error message is after starting the webwork container is:

```
app_1  | end fixing ownership and permissions
app_1  | 	(in cleanup) Not a GLOB reference at /opt/webwork/webwork2/lib/WeBWorK/CourseEnvironment.pm line 200.
app_1  | AH00526: Syntax error on line 54 of /etc/apache2/conf-enabled/webwork.conf:
app_1  | failed to read file /opt/webwork/courses/___/simple.conf: No such file or directory at /opt/webwork/webwork2/lib/WeBWorK/CourseEnvironment.pm line 186.\n

```

To reproduce this, change the first line in the `Dockerfile` from

`FROM ubuntu:16.04`
to
`FROM ubuntu:19.04`

Then execute 

```
docker-compose build
docker-compose up
```

This is not really a bug, as the Dockerfile specifies ubuntu 16.04, but I guess it would nevertheless be desirable to be able to run webwork on newer versions of Ubuntu.

The problem does not occur with

`FROM ubuntu:18.04`.

I remember having encountered this problem also using Debian buster on a bare metal server without using docker, but I did not confirm it.

**Comment by taniwallach** [comment]
The same error message about 
```
Not a GLOB reference at /opt/webwork/webwork2/lib/WeBWorK/CourseEnvironment.pm line 200.
```
was reported in the [forums](http://webwork.maa.org/moodle/mod/forum/discuss.php?d=4607) by  Jeremy Sylvestre. 

A workaround given there is to modify `lib/WeBWorK/CourseEnvironment.pm` in a particular way to bypass the specific "name" ("include") which triggered the problem. It seems that this is just some sort of reference to a local "include" function which does not need to be part of the final CourseEnvironment hash, so bypassing it is not likely to cause trouble.

@mgage commented there about the proposed change and that he did not foresee any adverse side-effect of the change, but did remark that the proposed change will need testing.

I'm starting to test the proposed change in my personal Docker setup, but not yet in production.

**Comment by taniwallach** [comment]
@heiderich - If you have a chance - maybe see if the proposed change solves the problem with using Ubuntu 19.04 as the base image for Docker. (Sorry, I don't want to mess with my Docker image at present.)

**Comment by heiderich** [comment]
@taniwallach Thank you for calling my attention to this post. I did some tests:

Making the change proposed in this forum post change, i.e. changing

https://github.com/openwebwork/webwork2/blob/075d8a8ac490fbfddd58f3612b232e82a7c4bb5b/lib/WeBWorK/CourseEnvironment.pm#L205

to

`next if $name =~ /^(INC|_.*|__ANON__|main::|include)$/;`

makes the error disappear. I tested this (using the develop branch) both with

`FROM ubuntu:16.04
`
and

`FROM ubuntu:19.04`

In both cases I did not experience the error anymore. I did not do further tests. But probably I will use this change in my docker test containers from now on.

It might indeed be related to the Perl version. Ubuntu 16.04 comes with 5.22, while 19.04 comes with 5.28.
